tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -39,16 +40,16 @@ public void ActivityStarted<T>(string sourceName, T activity)
 
             // Propagate Trace and Parent Span ids
             SpanContext? parent = null;
-            ulong? traceId = null;
-            ulong? spanId = null;
+            TraceId traceId = default;
+            ulong spanId = 0;
             string? rawTraceId = null;
             string? rawSpanId = null;
 
             if (activity is IW3CActivity w3cActivity)
             {
                 // If the user has specified a parent context, get the parent Datadog SpanContext
-                if (w3cActivity.ParentSpanId is not null
-                    && w3cActivity.ParentId is string parentId
+                if (w3cActivity.ParentSpanId != null!
```

Comment:
My precious `is` operator 😭

Reply:
Sorry! You don't get nullability warnings here? I changed this to `==` so I could suppress the warning with `!`.

Reply:
Hmm maybe VS just isn't yelling at me like Rider 🤔